service/storage_service.cc

```diff
@@ -1911,6 +1914,138 @@ future<> storage_service::decommission() {
     });
 }
 
+// Runs inside seastar::async context
+void storage_service::run_replace_ops() {
+    if (!db().local().get_replace_address()) {
+        throw std::runtime_error(format("replace_address is empty"));
+    }
+    auto replace_address = db().local().get_replace_address().value();
+    auto uuid = utils::make_random_uuid();
+    // TODO: Specify ignore_nodes
+    std::list<gms::inet_address> ignore_nodes;
+    // Step 1: Decide who needs to sync data for replace operation
+    std::list<gms::inet_address> sync_nodes;
+    for (const auto& x :_gossiper.endpoint_state_map) {
+        seastar::thread::maybe_yield();
+        const auto& node = x.first;
+        slogger.debug("replace[{}]: Check node={}, status={}", uuid, node, _gossiper.get_gossip_status(node));
+        if (node != get_broadcast_address() &&
+                node != replace_address &&
+                (_gossiper.is_normal(node) || _gossiper.is_shutdown(node)) &&
```

Comment:
I did not use get_live_token_owners  for various reasons. 1) we do not care about / want to depend on gossip liveness here because we will send verb to those nodes anyway 2) Check normal and shutdown status directly is better than checking is_dead_state. If a node is in the ring, it should be either normal or shutdown.

Reply:
Regardless of how the check should look like, we should extract it so that it's clear what we intend it to mean, e.g `is_ring_member()`. 

If we have that, then it's better if all the places which share the meaning reuse this function, because then there is no duplication of logic and there is a higher chance that the whole system remains correct now and in the future.

Reply:
ok. I will add a helper.